enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,307 @@
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2023-01-17
+last-updated: 2023-01-17
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-389
+see-also:
+  - "/enhancements/oc/must-gather.md"
+---
+
+# MicroShift supportability tools
+
+## Summary
+This enhancement proposes the addition of supportability tooling to MicroShift.
+
+To debug issues in the cluster we need a single command that any user or
+administrator is able to run to gather all the required information to diagnose
+and solve problems.
+
+## Motivation
+Most readers will be familiar with `must-gather` for OpenShift clusters and
+what it achieves. For MicroShift we need a tool with the same principles and
+goals, which is to gather every bit of information from the cluster to be able
+to diagnose and fix issues.
+
+### User Stories
+
+* As a MicroShift administrator, I want to be able to get a report about current
+  cluster status.
+* As a MicroShift developer, I want to be able to diagnose issues with a detailed
+  debug report.
+
+### Goals
+* Provide a single tool to automate the report creation.
+
+### Non-Goals
+N/A
+
+## Proposal
+While `must-gather` tool is a widely known and used for debugging clusters, it
+may not be the most suitable for MicroShift use cases, and requires deep
+modifications. The existing `must-gather` implementation:
+1. Needs many OpenShift APIs present, like `clusteroperators`.
+1. Expects certain OpenShift components to be present, like
+machine-config-daemon.
+1. Runs commands within a pod, which means pulling another container over
+network (or pre-caching it).
+1. We want the debugging workflow for the product to be consistent, regardless
+of what is running on it and sos pulls info from many components not present in
+MicroShift.
+
+Some of these reasons are a no-go for MicroShift because of constrained
+environments. Some others mean there should be significant changes to
+`must-gather` to have MicroShift awareness.
+
+There is another tool for gathering info from a running RHEL system, and that
+is [`sos`](https://sos.readthedocs.org/). This tool is present in RHEL repos
+and is capable of generating reports for many components and applications in
+the system by using plugins. By executing `sos report` it will generate a
+detailed report with all the available information it can extract.
+
+Plugins have an enablement check or may be invoked manually when executing
+the command. A [plugin](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin)
+is a python script that we include as part of `sos`.
+
+There is a [plugin for OpenShift](https://github.com/sosreport/sos/blob/main/sos/report/plugins/openshift.py)
+already that almost fits the needs of this EP. MicroShift does not share the
+same resources, APIs and namespaces/names that the plugin assumes, as it was
+made for OpenShift clusters. However, much of the needs covered in the plugin
+are the same, so the contents of the plugin may be used too.
```

Comment:
Actually the plugin pulls some core information. However, before you mentioned that the must-gather collects the machine configs, which is not there. Just FYI< sosreport does the same thing. https://github.com/sosreport/sos/blob/main/sos/report/plugins/openshift.py#L265-L286

Reply:
@pacevedom , we should make clear that the plan is to create a new plugin based on the existing openshift plugin, but not to use that plugin directly.

Reply:
@dhellmann If I would recommend, if new plugin should be created, it should be `openshift_microshift.py` pointing that this is a microshift installation. The original plugin uses the `oc` binary, where in the specific plugin, it would use `microshift` binary. Please add to the enhancement the list of info you need to collect.

Reply:
We can consider that name. In the user documentation we are trying to de-emphasize the relationship between OpenShift and MicroShift because of the huge difference in features, so I'd prefer to just call it "microshift".

Including the set of info we will collect in the enhancement is a good idea.